hnn_core/network_models.py

```diff
@@ -11,7 +11,8 @@
 from .externals.mne import _validate_type
 
 
-def jones_2009_model(params=None, add_drives_from_params=False):
+def jones_2009_model(params=None, add_drives_from_params=False,
+                     legacy_mode=True):
```

Comment:
why did you add this here? We need to set ourselves free of the legacy ...

Reply:
I promise it's actually towards that goal! `legacy_mode=True` by default in the `Network`. So the old interface for `jones_2009_model()` didn't have the option to turn it on or off.

The reason I added it is because `legacy_mode` needs to be `False` to add a drive that targets a specific section. This is because `legacy_mode=True` creates a connection for _all_ cells, even if they aren't specified in  the weights dict (they just have a weight of zero). Therefore if you try to target `apical_trunk` for example, it'll throw an error because it tries to create a connection to the basket cells when they don't have a section named `apical_trunk`.

Reply:
Would a good fix be to just make that clearer in the docstring?

Reply:
If you plan to deprecate it in the future, the default should be `False` and it should raise a deprecation warning if someone sets it to `True`.

Reply:
It sounds like the examples are also using `legacy_mode=True` ? I think the "right" thing to do is to start a separate PR that fixes this problem, then rebase this PR after that is merged.

Reply:
If we set default to `False` here, then we'll need to explicitly set `legacy_mode=True` for all of our examples + tests that use this network model in order to maintain consistency. Either that or we just rip the bandaid off and change the outcome of our examples + tests now.

Reply:
Btw, I noticed that the new GUI currently loads the "ghost" drives ... probably because of the implicit `legacy_mode=True`

Reply:
Ugh. Legacy mode is going to be the death of us.

Reply:
@jasmainak that's exactly what's happening. Considering this change doesn't require changes to the current code, can we deal with `legacy_mode` in a separate PR?